src/client/Microsoft.Identity.Client.Broker/BrokerExtension.cs

```diff
@@ -47,29 +47,55 @@ public static PublicClientApplicationBuilder WithBrokerPreview(this PublicClient
         /// parameters, and to create a public client application instance</returns>
         public static PublicClientApplicationBuilder WithBroker(this PublicClientApplicationBuilder builder, BrokerOptions brokerOptions)
         {
-            AddRuntimeSupportForWam(builder);
+            AddRuntimeSupport(builder, true);
             builder.Config.BrokerOptions = brokerOptions;
             builder.Config.IsBrokerEnabled = brokerOptions.IsBrokerEnabledOnCurrentOs();
             return builder;
         }
 
-        private static void AddRuntimeSupportForWam(PublicClientApplicationBuilder builder)
+        /// <summary>
+        /// Use this API to enable SsoPolicy enforcement when not using broker.
+        /// 
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns>A <see cref="PublicClientApplicationBuilder"/> from which to set more
+        /// parameters, and to create a public client application instance</returns>
+        public static PublicClientApplicationBuilder WithSsoPolicy(this PublicClientApplicationBuilder builder)
+        {
+            AddRuntimeSupport(builder, false);
+            builder.Config.IsSsoPolicyEnabled = true;
+            return builder;
+        }
+
+        private static void AddRuntimeSupport(PublicClientApplicationBuilder builder, bool isWamBrokerScenario)
         {
             if (DesktopOsHelper.IsWin10OrServerEquivalent())
```

Comment:
So we only care about 1P apps on Windows?

Reply:
Yes thats the plan for now. Runtime does support cross plat and if there is need for this in other platforms, we can extend it